articles/introduction-to-processors/index.md

```diff
@@ -0,0 +1,89 @@
+However, computers are binary and can only “understand” data in two states. These are usually described as on & off or 1 & 0. 
+
+Using high-level languages such as python and java, programmers pass instructions to processors to accomplish certain tasks. The code is then passed on to either an assembler, compiler, or interpreter. These are the programs that translate code from human language to assembly or directly to machine language. Assembly language is a low-level language that resembles human language. Assembly has a low level of __abstraction__ and therefore runs very fast.
+
+> Abstraction refers to the hiding of detail that “protects” the programmer from the complexities of the particulars. 
+
+![alt text](Abstraction.png)
+
+Let us consider the printing of ‘Hello World’ at various levels.  C++ has a high level of abstraction as few lines of code are required to complete the task. Machine Language does not hide any details of the process, and thus translation is not required. Consequently, machine language would run fastest. As a middle ground, Assembly language helps programmers capitalize on speed and code efficiency as they understand the translation process better. 
+### Hello World Program
+| C++  | 
+| ------------- |
+```
+#include <iostream>
+
+int main()
+{
+    
+    std::cout << "Hello, world!\n";
+    return 0;
+}
+ ```    
+
+| Assembly Language | 
+| ------------- |
+```
+global  _main
+    extern  _printf
+
+    section .text
+_main:
+    push    message
+    call    _printf
+    add     esp, 4
+    ret
+message:
+    db  'Hello, World', 10, 0
+```
+
+| Machine Language | 
+| ------------- |
+```
+b8    21 0a 00 00   #moving "!\n" into eax
+a3    0c 10 00 06   #moving eax into first memory location
+b8    6f 72 6c 64   #moving "orld" into eax
+a3    08 10 00 06   #moving eax into next memory location
+b8    6f 2c 20 57   #moving "o, W" into eax
+a3    04 10 00 06   #moving eax into next memory location
+b8    48 65 6c 6c   #moving "Hell" into eax
+a3    00 10 00 06   #moving eax into next memory location
+b9    00 10 00 06   #moving pointer to start of memory location into ecx
+ba    10 00 00 00   #moving string size into edx
+bb    01 00 00 00   #moving "stdout" number to ebx
+b8    04 00 00 00   #moving "print out" syscall number to eax
+cd    80            #calling the linux kernel to execute our print to stdout
+b8    01 00 00 00   #moving "sys_exit" call number to eax
+cd    80            #executing it via linux sys_call
+```
+
+
+
+To understand how code is turned from human-readable to machine-readable then executed, we shall use the __x86 Instruction Set Architecture__. This refers to a series of microprocessors developed by Intel based on the 8085 Processor. A __microprocessor__ is a programmable electronic chip that has computing and decision-making capabilities. These microprocessors(x86) share similar instruction set architectural models and are hence grouped. For example: 80186 & 80286 microprocessors.

```

Comment:
Can you elaborate on the x86 instruction set?Additional information is required.